### PR TITLE
Add secret store section to Developer docs

### PR DESCRIPTION
**Pull request by willtsai**
Thank you for helping make the Radius documentation better!

**Please follow this checklist before submitting:**

- [x] [Read the contribution guide](https://docs.radapp.dev/contributing/docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

- Edited Radius diagrams to include Secret Store resource
- Added Secret Store developer documentation

### Summary
📄🧩🔒

Added documentation and code snippets for secrets management in Radius applications. The new page `docs/content/author-apps/secrets/_index.md` explains how to use the `Applications.Core/secretStores` resource to store or reference secrets of type certificate.

> _`secretStores` bicep_
> _New doc page for secrets_
> _Autumn leaves secure_

### Walkthrough
* Add a new documentation page for secrets management in Radius applications ([link](https://github.com/project-radius/docs/pull/607/files?diff=unified&w=0#diff-ef45b60b9096370b2cbac5fa24cb5f3fc203601d8cbb6419fb43b0567fc3f604R1-R32))
  * Provide code examples using the `Applications.Core/secretStores` resource in bicep ([link](https://github.com/project-radius/docs/pull/607/files?diff=unified&w=0#diff-b156c89db7c4a63755da000e02ac7a1406ac76fb0e164b52fe5c20f3e41d26fcR1-R61))
* Add a new bicep snippet file for the secret store documentation page ([link](https://github.com/project-radius/docs/pull/607/files?diff=unified&w=0#diff-b156c89db7c4a63755da000e02ac7a1406ac76fb0e164b52fe5c20f3e41d26fcR1-R61))

## Issue reference

Fixes: https://github.com/project-radius/radius/issues/2310

### CODE REVIEW

File: docs/content/author-apps/secrets/_index.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+Here is an example for creating a new Secret Store resource and storing a TLS certificate in it. Radius leverages the secrets management solution available on the hosting platform to create and store the secret. For example, if you are deploying to Kubernetes, the secret will be created in Kubernetes Secrets.
+
+{{< rad file="snippets/secretstore.bicep" embed=true marker="//SECRET_STORE_NEW" >}}
+
+## Reference an existing Secret Store
+
+Here is an example of using a Secret Store to reference secrets stores in an existing secrets management solution that is external to the Radius application stack. Note that only references to Kubernetes Secrets is currently supported, with more to come in the future.
+
+{{< rad file="snippets/secretstore.bicep" embed=true marker="//SECRET_STORE_REF" >}}
+
```
Comment (AaronCrawfis) [inline docs/content/author-apps/secrets/_index.md:28]:
```suggestion

## Using Secret Stores

Secret Stores can currently be used for TLS certificates with [Radius Gateways]({{< ref "networking#gateways" >}}).

Additional use-cases will be added in upcoming releases.

```

File: docs/content/author-apps/secrets/snippets/secretstore.bicep [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    resource: 'secret-app-existing-secret' // Reference to the name of an external secret store
+    type: 'certificate' // The type of secret in your resource
+    data: {
+      // The keys in this object are the names of the secrets in an external secret store
+      'tls.crt': {}
+      'tls.key': {}
+    }
+  }
+}
+//SECRET_STORE_REF
```
Comment (AaronCrawfis) [inline docs/content/author-apps/secrets/snippets/secretstore.bicep]:
```suggestion
import radius as radius

@description('The ID of your Radius environment. Injected automatically by the rad CLI.')
param environment string

@description('The data for your TLS certificate')
@secure()
param tlscrt string

@description('The key for your TLS certificate')
@secure()
param tlskey string

resource app 'Applications.Core/applications@2022-03-15-privatepreview' = {
  name: 'corerp-resources-secretstore'
  properties: {
    environment: environment
  }
}

//SECRET_STORE_NEW
resource appCert 'Applications.Core/secretStores@2022-03-15-privatepreview' = {
  name: 'appcert'
  properties:{
    application: app.id
    type: 'certificate'
    data: {
      'tls.key': {
        value: tlskey
      }
      'tls.crt': {
        value: tlscrt
      }
    }
  }
}
//SECRET_STORE_NEW

//SECRET_STORE_REF
resource existingAppCert 'Applications.Core/secretStores@2022-03-15-privatepreview' = {
  name: 'existing-appcert'
  properties:{
    application: app.id
    resource: 'secret-app-existing-secret' // Reference to the name of an external Kubernetes secret store
    type: 'certificate' // The type of secret in your resource
    data: {
      // The keys in this object are the names of the secrets in an external secret store
      'tls.crt': {}
      'tls.key': {}
    }
  }
}
//SECRET_STORE_REF
```

**Review by AaronCrawfis — CHANGES_REQUESTED** [review]
Left a couple small comments

**Comment by willtsai** [comment]
> Left a couple small comments

committed your suggestions, thanks! please take a look again.

**Review by AaronCrawfis — APPROVED** [review]

**Review by AaronCrawfis — APPROVED** [review]

**Pull request merged**